### ISSUE TITLE
IV falsely detected as not properly generated

### ISSUE THREAD

**Issue by akwick** [CROSSINGTUD/CryptoAnalysis#68]
# Steps done:
* Applying CryptoAnalysis on the project [dragonite](https://github.com/dragonite-network/dragonite-java) with commit id e26ffff0b.
* The analysis reports 5 violations (2 RequiredPredicateError and 3 IncompleOperationErrorr). More Details in the [attached report](https://github.com/CROSSINGTUD/CryptoAnalysis/files/2302996/CogniCrypt-Report.dragonite-java.txt)

* For the RequiredPredicateError violation for method *encryptImpl*, we assume that the report is a false positive.
```
in Method: byte[] encryptImpl(byte[])
                RequiredPredicateError violating CrySL rule for IvParameterSpec
                        First parameter was not properly randomized
                        at statement: specialinvoke $r6.<javax.crypto.spec.IvParameterSpec: void <init>(byte[])>(r16)
```
* The reported violation is in the file *AESCrypto.java* in line 95. 

# Notes why we assume that the violation is a false positive

- encryptioCipher mode is choosen secure.
- Ins okay by calling SecureRandom in line 50
 - Seeds is optional -> not called
- Ends - nextBytes in line 93 is called -> Nexts / Ends is done-> ensures randomized[next] which is used in IvParameterSpec as parameter.

![20180815_dragonite_cryslrule](https://user-images.githubusercontent.com/4615245/44350619-1fcc3780-a4a0-11e8-80d9-28a6a1fa2b9d.PNG)

/cc @anam-dodhy

**Comment by johspaeth** [comment]
I tried to reproduce the finding on a small code examples in the following classes:

- [x] [issue68.simplified.nofield.AESCryptor](https://github.com/CROSSINGTUD/CryptoAnalysis/commit/c1d44fc1fedd74f77fbf7e5eefff26a60db10f36#diff-77d710bcbcc95acb6179d7d17ae2dd98): Not using SecureRandom as a field, instead as a local variable. No finding :+1: 
- [x] [issue68.simplified.field.AESCryptor](https://github.com/CROSSINGTUD/CryptoAnalysis/commit/c1d44fc1fedd74f77fbf7e5eefff26a60db10f36#diff-00a0740a5f27f4df344a3c6ec0b73a1f): Using SecureRandom in a field. No finding :+1: 
- [x] [issue68.AESCryptor](https://github.com/CROSSINGTUD/CryptoAnalysis/commit/c1d44fc1fedd74f77fbf7e5eefff26a60db10f36#diff-5bf56afc33b326b3f990cc3501564011): Copy of the whole class. No finding :+1:
- [ ] Instantiate analysis on whole project, to see where the error originates from.

**Comment by johspaeth** [comment]
@akwick @anam-dodhy
Does anyone have a readily compiled .jar that we could use to run the analysis another time?

**Comment by anam-dodhy** [comment]
I have pushed the build files of the project "dragonite-java" at the following repository [anam-dodhy/dragonite-java](https://github.com/anam-dodhy/dragonite-java) and I originally ran CryptoAnalysis on the build files present at the following path of this project 
"dragonite-java/dragonite-sdk/build/classes/java/main/"
Moreover, I have not tried to reproduce this error with the development version of CryptoAnalysis.

**Comment by johspaeth** [comment]
I have done my best to fix it and close this issue now.
